app/src/main/java/agency/five/codebase/android/movieapp/ui/favorites/FavoritesViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package agency.five.codebase.android.movieapp.ui.favorites
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.ui.favorites.mapper.FavoritesMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class FavoritesViewModel(
+    private val movieRepository: MovieRepository,
+    favoritesMapper: FavoritesMapper,
+) : ViewModel() {
+    val favoritesViewState: StateFlow<FavoritesViewState> = movieRepository.favoriteMovies()
+        .map { movies -> favoritesMapper.toFavoritesViewState(movies) }
+        .stateIn(viewModelScope, SharingStarted.Eagerly, FavoritesViewState(emptyList()))
+
+    fun toggleFavorite(movieId: Int) {
```

Comment:
Since you can only remove movies from favorites on this this screen this method should be renamed and it should call a more appropriate method from `movieRepository`